include/RevSysCalls.h

```diff
@@ -41,7 +41,7 @@ struct EcallState {
     string.clear();
     path_string.clear();
     bytesRead = 0;
-    buf[0] = '\0';
+    buf.fill('\0');
```

Comment:
I don't think we have to `fill` the entire `buf` -- or even the first byte of it, with `\0`. In `rev_write` we use:
```c++
    EcallState.string += std::string_view(EcallState.buf.data(), EcallState.bytesRead);
```
The `buf.data()` returns a pointer to the underlying array data, and `bytesRead` tells how many bytes we've read so far, so clearing `bytesRead` should be sufficient.

The `string_view` creates a view of a not-necessarily-null-terminated string which is a raw number of bytes. That way we can read data containing `\0` and not stop on `\0`. Although `std::string` is guaranteed to be null-terminated, it may contain `\0` bytes embedded within its first `size()` bytes.